api/oc_assert.c

```diff
@@ -0,0 +1,33 @@
+/******************************************************************
+ *
+ * Copyright (c) 2023 plgd.dev s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"),
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ******************************************************************/
+
+/**
+ * @file oc_assert.c
+ *
+ * @author Jozef Kralik, Daniel Adam
+ */
+
+#include "port/oc_assert.h"
+#include "port/oc_log_internal.h"
+
+void
+oc_abort(const char *msg)
+{
+  PRINT("\n%s\nAbort.\n", msg);
```

Comment:
I think we can move the `PRINT` macro into the public oc_log.h header, it's used in the apps examples, which should use only public stuff.